resources-manager/include/SumsResourceEntitiesAvailabilityManager.hpp

```diff
@@ -0,0 +1,44 @@
+// Concord
+//
+// Copyright (c) 2018-2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#pragma once
+
+#include "IResourceManager.hpp"
+#include "ISystemResourceEntity.hpp"
+
+#include <vector>
+#include <memory>
+#include <algorithm>
+
+namespace concord::performance {
+
+class SumsResourceEntitiesAvailabilityManager : public IResourceManager {
```

Comment:
Do you have a diagram that shows the relationship between those classes and how they interact in order to calculate a result?
For example, I don't understand SumsResourceEntitiesAvailabilityManager and SubstractFromMaxResourceManager and where and how they are being used.

Reply:
So far the only usage is in the unittests. The responsiblity of computations is on the implementation of the IResourceManager interface. The SumsResourceEntitiesAvailabilityManager sums up the getAvailableResources answers.
The SubstractFromMaxResourceManager  substracts from configrued maximum capacity the sum of getMeasurements.

Reply:
More explanations are in attached pdf file.

Reply:
[ResourceManagerScenarios.pdf](https://github.com/vmware/concord-bft/files/7884184/ResourceManagerScenarios.pdf)